centreon/src/Centreon/Domain/Contact/Contact.php

```diff
@@ -119,9 +119,9 @@ class Contact implements UserInterface, ContactInterface
     public const ROLE_CONFIGURATION_POLLERS_AGENT_CONFIGURATIONS_RW = 'ROLE_CONFIGURATION_POLLERS_AGENT_CONFIGURATIONS_RW';
 
     /**
-     * @var string
+     * @var string|null
      */
-    public const DEFAULT_LOCALE = 'en_US';
+    public const DEFAULT_LOCALE = null;
```

Comment:
I have some doubts on the location for the fix.
At this level (Domain/Contact) I suspect we want to keep the default locale as it was and prefer to apply the change directly in the subscriber instead to avoid edge effect

https://github.com/centreon/centreon/blob/da8f6abe3afc11e7850cb1bf8454d70f622b55f4/centreon/src/EventSubscriber/CentreonEventSubscriber.php#L453-L458

My suggestion would be no change in Contact & Repository.
Only in the subscriber

```php
return isset($_SERVER['HTTP_ACCEPT_LANGUAGE']) 
     ? (string) \Locale::acceptFromHttp($_SERVER['HTTP_ACCEPT_LANGUAGE']) 
     : null; 
```

Reply:
Will try it out

Reply:
<img width="194" alt="Screenshot 2025-05-16 at 6 57 02 PM" src="https://github.com/user-attachments/assets/abf84dbc-a81f-4c92-8295-f54390b4d564" />
It does not work

Reply:
It figured our why 

<img width="583" alt="Screenshot 2025-05-16 at 6 59 16 PM" src="https://github.com/user-attachments/assets/870e0e55-9b10-4cf6-b37a-ca8ce3a8ed81" />

It is because the user->getLocale() is 'en_US' by default so it is defined so it does not execute $this->getBrowserLocale()

Reply:
For what I've tested it does not break anything in the code

Reply:
And this is the only usage of getBrowserLocale in the code

Reply:
```suggestion
    public const DEFAULT_LOCALE = 'en_US';
```